Shared/Tests.Shared/RelationshipTests.cs

```diff
@@ -479,6 +479,23 @@ public void Insert_WhenIndexIsEqualToCount_ShouldWork()
             Assert.That(person.Dogs.Count, Is.EqualTo(2));
         }
 
+        [Test]
+        public void Backlinks()
```

Comment:
I'm not sure it matters much at runtime. We only do relationship validation in the weaver so there are tests for both kinds of relationships there. In runtime we're basically calling through to core and our code doesn't care whether the other end of a relationship is `Object` or `Array` in the schema.